### ISSUE TITLE
About the final likelihood calculation.

### ISSUE THREAD

**Issue by tu-curious** [christegho/bnn-mnist#1]
https://github.com/christegho/bnn-mnist/blob/7f71c0502ec1bec283941baa1d9fde8008ad5761/bbn.py#L120\

Hi @christegho ,

I had a question about this line (120). As per my understanding, the variables y and h are sparse categorical variables and they should have a categorical likelihood (\sum(y*log(y_{hat}))) instead of the Gaussian likelihood. Please let me know if my understanding is wrong.

If you think I am correct, please have a look at my repo (inspired from your repo, one other page and the paper itself) for correctness.
https://github.com/tushar-agarwal2909/Bayes_by_Backprop-TF

Thanks

**Comment by christegho** [comment]
Hi Tushar,

I think you are correct in minimising a categorical loss in the objective. Taking a log Gaussian is equivalent to doing MSE. In the case of MNIST labels, a categorical loss is more fit. I will change my code to use something similar to what you did with `tf.reduce_sum(log_categ(y, y_hat))`.

I looked at your code, I'd say make sure you are minimising the cross entropy loss. In your optimisation, you are minimising the negative categorical loss (i.e. maximising it). (When you minimise the negative log Gaussian (<=> maximising the log likelihood), you are in fact minimising the negative of the negative of the MSE (there is a minus sign before the (x-mean)^2 in the Gaussian, then a minus sign before the log Gaussian, and we minimise all of this).

**Comment by christegho** [comment]
This was resolved in 
<https://github.com/christegho/bnn-mnist/pull/2>
